test-resources/headless-core-stub/lambdas/callback/src/callback-handler.ts

```diff
@@ -1,12 +1,87 @@
-        return {
-            status: "200",
-            body: "Hello callback",
-        };
+    async handler(event: APIGatewayProxyEvent, _context: Context): Promise<APIGatewayProxyResult> {
+        try {
+            const authorizationCode = event.queryStringParameters?.authorizationCode;
+
+            if (!authorizationCode) {
+                return this.respondWith(400, "Missing authorization code");
+            }
+            logger.info(`Received authorizationCode: ${authorizationCode}`);
+
+            logger.info("Fetching session item...");
+            const sessionItem = await callbackService.getSessionByAuthorizationCode(
+                sessionTableName,
+                authorizationCode,
+            );
+
+            logger.info("Fetching SSM parameters");
+            const ssmParameters = await configurationHelper.getParameters(sessionItem.clientId);
+            const privateJwtKey = JSON.parse(ssmParameters["privateSigningKey"]);
+            const audience = ssmParameters["audience"];
+
+            logger.info("Generating private JWT parameters...");
+            const privateJwtParams = await generatePrivateJwtParams(
+                sessionItem.clientId,
+                authorizationCode,
+                sessionItem.redirectUri,
+                privateJwtKey,
+                audience,
+            );
+
+            const audienceApi = this.formatAudience(audience);
+            logger.info(`Audience is ${audienceApi}`);
+
+            const tokenEndpoint = `${audienceApi}/token`;
+
+            logger.info(`Calling token endpoint ${tokenEndpoint} with body: ${privateJwtParams}`);
+            const tokenResponse = await callbackService.getToken(tokenEndpoint, privateJwtParams);
+
+            if (!tokenResponse.ok) {
+                const tokenResponseBody = await tokenResponse.text();
+                this.logApiError(tokenEndpoint, tokenResponse.status, tokenResponseBody);
+                return this.respondWith(tokenResponse.status, tokenResponseBody);
+            }
+
+            logger.info("Successfully called /token endpoint");
+            const tokenBody = await tokenResponse.json();
+
+            const credentialEndpoint = `${audienceApi}/credential/issue`;
+            logger.info(`Calling ${credentialEndpoint}`);
+            const credential = await callbackService.issueCredential(credentialEndpoint, tokenBody.access_token);
+            const credentialResponseBody = await credential.text();
+
+            if (!credential.ok) {
+                this.logApiError(credentialEndpoint, credential.status, credentialResponseBody);
+            }
+            logger.info("Successfully called /credential/issue endpoint");
+
+            return this.respondWith(credential.status, credentialResponseBody);
+        } catch (error: unknown) {
+            const err = error as Error;
+            logger.error(err.message);
+            return this.respondWith(500, err.message);
```

Comment:
Similar comment about the error handling - the /start PR has a slightly different pattern so let's make sure they're aligned. 

Also are we sure we definitely need this `respondWith` abstraction? Looks like it just formats an object which is pretty simple to read on it's own I think?